hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -134,26 +150,36 @@ public void doZeroSum(final List<BalanceChange> changes) {
             }
         }
 
+        if (validity == OK
+                && (autoCreationFee > 0)
+                && (autoCreationFee > (long) accountsLedger.get(txnCtx.activePayer(), BALANCE))) {
+            validity = INSUFFICIENT_PAYER_BALANCE;
+        }
+
         if (validity == OK) {
             adjustBalancesAndAllowances(changes);
             if (autoCreationFee > 0) {
-                payFunding(autoCreationFee);
+                payAutoCreationFee(autoCreationFee);
                 autoCreationLogic.submitRecordsTo(recordsHistorian);
+                autoCreationLogic.clearTokenAliasMap();
             }
         } else {
             dropTokenChanges(sideEffectsTracker, nftsLedger, accountsLedger, tokenRelsLedger);
             if (autoCreationLogic != null && autoCreationLogic.reclaimPendingAliases()) {
                 accountsLedger.undoCreations();
+                autoCreationLogic.clearTokenAliasMap();
```

Comment:
Can also be deleted once we clear in `AutoCreationLogic.reset()`.

Reply:
Done !